### PR TITLE
Making tags requirement same as those in Kubernetes

### PR DESCRIPTION
**Pull request by mbanikazemi**
Changing validation check to make sure non-empty tags start with an
alphanumeric character

### CODE REVIEW

File: pilot/pkg/model/validation.go [hunk 1]
```diff
@@ -43,8 +43,10 @@ const (
 	// a wild-card prefix is an '*', a normal DNS1123 label with a leading '*' or '*-', or a normal DNS1123 label
 	wildcardPrefix string = `\*|(\*|\*-)?(` + dns1123LabelFmt + `)`
 
-	// TODO: there is a stricter regex for the labels from validation.go in k8s
-	qualifiedNameFmt string = "[-A-Za-z0-9_./]*"
+	// Using kubernetes requirement, a valid tag must be an empty string or consist
+	// of alphanumeric characters, '-', '_' or '.', and must start and end with an
+	// alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345'
+	qualifiedNameFmt string = "(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?"
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/validation.go]: IN k8s it’s
```
const qnameCharFmt string = "[A-Za-z0-9]"
const qnameExtCharFmt string = "[-A-Za-z0-9_.]"
const qualifiedNameFmt string = "(" + qnameCharFmt + qnameExtCharFmt + "*)?" + qnameCharFmt
```

Not the same as above

File: pilot/pkg/model/validation_test.go [hunk 2]
```diff
@@ -318,6 +318,14 @@ func TestLabelsValidate(t *testing.T) {
...
 			valid: true,
 		},
+		{
+			name: "bad tag",
+			tags: Labels{"_key": "value"},
+		},
+		{
+			name: "bad tag",
+			tags: Labels{"key": ".value"},
+		},
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/validation_test.go]: Add more cases, like `key_` and also some invalid characters

File: pilot/pkg/model/validation.go (hunk 1 above)
Comment (mbanikazemi) [reply pilot/pkg/model/validation.go]: Yes, the difference is that one allows empty strings and the other does not. Made changes to allow empty strings for label values and not label keys. This is similarly done in Kubernetes with labelValueFmt.

File: pilot/pkg/model/validation_test.go (hunk 2 above)
Comment (mbanikazemi) [reply pilot/pkg/model/validation_test.go]: Done.

**Comment by hzxuzhonghu** [comment]
/approve

leaves to @rshriram for another eye

**Pull request merged**